src/state/apis/foxy/foxyApi.ts

```diff
@@ -220,6 +220,7 @@ export const foxyApi = createApi({
 
         // RTK caches queries from inputs, thus re-calling this query for the same opportunity will return the cache data if not invalidated
         const accountFilter = { accountId }
+        const accountAddress = fromAccountId(accountId).account
```

Comment:
The other guts of this PR, doesn't use the passed `userAddress` anymore but makes it from the `accountId`, ensuring we get programmatic on accountId change instead of keeping both in sync